src/components/ValidatedInput/ValidFieldPassword.tsx

```diff
@@ -124,28 +119,64 @@ export class ValidFieldPassword extends React.Component<ValidFieldPasswordProps,
   };
 
   handleGenerateText = () => {
-    const { hasGeneratePassword, strengthLevelIndicator, customField } = this.props;
-    const { levelColor } = this.state;
+    const { hasGeneratePassword, customField, hasPasswordStrenghtMeter } = this.props;
+    const { passwordStrenghtLevel } = this.state;
     return (
       <>
+        {hasPasswordStrenghtMeter && passwordStrenghtLevel && (
+          <div className="password-strength-meter">
+            <div className={classNames("strength-meter-bar", passwordStrenghtLevel)} />
+            <span className={classNames("strength-meter-label")}>{translation("password.strength.label")}</span>
+            <span className={classNames("strength-meter-level", passwordStrenghtLevel)}>
+              {translation(`password.strength.level.${passwordStrenghtLevel}`)}
+            </span>
+          </div>
+        )}
         {hasGeneratePassword && (
           <Button appearance="link" spacing="none" className="generate-password" onClick={this.handleGeneratePassword}>
             {translation("password.generateNew")}
           </Button>
         )}
-        {strengthLevelIndicator && (
-          <span className={classNames("level", levelColor)}>{translation(`password.${strengthLevelIndicator}`)}</span>
-        )}
         {customField}
       </>
     );
   };
 
+  calculatePasswordStrenght = (password: string) => {
+    const { score } = zxcvbn(password);
+    switch (score) {
+      case 0:
+        this.setState({ passwordStrenghtLevel: PASSWORD_STRENGTH_METER["very-weak"] });
+        break;
+      case 1:
+        this.setState({ passwordStrenghtLevel: PASSWORD_STRENGTH_METER["weak"] });
+        break;
+      case 2:
+        this.setState({ passwordStrenghtLevel: PASSWORD_STRENGTH_METER["fair"] });
+        break;
+      case 3:
+        this.setState({ passwordStrenghtLevel: PASSWORD_STRENGTH_METER["strong"] });
+        break;
+      case 4:
+        this.setState({ passwordStrenghtLevel: PASSWORD_STRENGTH_METER["very-strong"] });
+        break;
+    }
+  };
+
+  onChangeWrapper = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { onChange, hasPasswordStrenghtMeter } = this.props;
+    if (onChange) onChange(event);
+    if (!hasPasswordStrenghtMeter) return;
+    if (!event.target.value) return this.setState({ passwordStrenghtLevel: null });
+    this.calculatePasswordStrenght(event.target.value);
```

Comment:
please consider this [issue](https://github.com/dropbox/zxcvbn/issues/316). seems like we need to debounce calculate password strength

Reply:
I have mention this on the kick off meeting and it seems only happen on that special character being copy paste some time, it seems fine for now we can update the packages when they fix it.. 

also I will try the [zxcvbn-ts](https://www.npmjs.com/package/@zxcvbn-ts/core) that you mention hopefully they don't have this bug

Reply:
unfortunately this issue still happen at [zxcvbn-ts](https://www.npmjs.com/package/@zxcvbn-ts/core)  I'm adding debounce